// plugin-welcome/src/api/models/ControllersPersonaldata.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersPersonaldata
 */
export interface ControllersPersonaldata {
    /**
     * 
     * @type {string}
     * @memberof ControllersPersonaldata
     */
    personalMail?: string;
    /**
     * 
     * @type {string}
     * @memberof ControllersPersonaldata
     */
    personalName?: string;
    /**
     * 
     * @type {string}
     * @memberof ControllersPersonaldata
     */
    personalPhone?: string;
}

export function ControllersPersonaldataFromJSON(json: any): ControllersPersonaldata {
    return ControllersPersonaldataFromJSONTyped(json, false);
}

export function ControllersPersonaldataFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersPersonaldata {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'personalMail': !exists(json, 'personalMail') ? undefined : json['personalMail'],
        'personalName': !exists(json, 'personalName') ? undefined : json['personalName'],
        'personalPhone': !exists(json, 'personalPhone') ? undefined : json['personalPhone'],
    };
}

export function ControllersPersonaldataToJSON(value?: ControllersPersonaldata | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'personalMail': value.personalMail,
        'personalName': value.personalName,
        'personalPhone': value.personalPhone,
    };
}
