// plugin-welcome/src/api/models/ControllersSystemmember.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersSystemmember
 */
export interface ControllersSystemmember {
    /**
     * 
     * @type {string}
     * @memberof ControllersSystemmember
     */
    mail?: string;
    /**
     * 
     * @type {string}
     * @memberof ControllersSystemmember
     */
    password?: string;
}

export function ControllersSystemmemberFromJSON(json: any): ControllersSystemmember {
    return ControllersSystemmemberFromJSONTyped(json, false);
}

export function ControllersSystemmemberFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersSystemmember {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'mail': !exists(json, 'mail') ? undefined : json['mail'],
        'password': !exists(json, 'password') ? undefined : json['password'],
    };
}

export function ControllersSystemmemberToJSON(value?: ControllersSystemmember | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'mail': value.mail,
        'password': value.password,
    };
}
